watcher/lib/influxdb/rollups.go

```diff
@@ -126,29 +126,56 @@ func buildFunction(f Function) (string, error) {
 		alias = f.Field
 	}
 	if strings.HasPrefix(f.Function, constants.FunctionPercentile) {
-		value, err := parsePercentileValue(f.Function)
+		value, err := parseValueComposedFunc(f.Function)
 		if err != nil {
 			return "", trace.Wrap(err)
 		}
 		return fmt.Sprintf("%v(%v, %v) as %v", constants.FunctionPercentile, f.Field, value, alias), nil
 	}
+	if strings.HasPrefix(f.Function, constants.FunctionBottom) {
+		value, err := parseValueComposedFunc(f.Function)
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		return fmt.Sprintf("%v(%v, %v) as %v", constants.FunctionBottom, f.Field, value, alias), nil
+	}
+	if strings.HasPrefix(f.Function, constants.FunctionTop) {
+		value, err := parseValueComposedFunc(f.Function)
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		return fmt.Sprintf("%v(%v, %v) as %v", constants.FunctionTop, f.Field, value, alias), nil
+	}
+	if strings.HasPrefix(f.Function, constants.FunctionSample) {
+		value, err := parseValueComposedFunc(f.Function)
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		return fmt.Sprintf("%v(%v, %v) as %v", constants.FunctionSample, f.Field, value, alias), nil
+	}
 	return fmt.Sprintf("%v(%v) as %v", f.Function, f.Field, alias), nil
 }
 
-// parsePercentileValue parses the percentile value from the strings like "percentile_90"
-func parsePercentileValue(data string) (string, error) {
+// parseValueComposedFunc parses the additional arg value from the composed functions strings like "percentile_90"
+func parseValueComposedFunc(data string) (string, error) {
 	parts := strings.Split(data, "_")
 	if len(parts) != 2 {
 		return "", trace.BadParameter(
-			"percentile function must have format like 'percentile_90'")
+			"percentile function must have format like 'percentile_90', 'top_10', 'bottom_10' or 'sample_1000' ")
```

Comment:
This is no longer related to the `percentile` function. See above for an abbreviated version of this code.